### DISCUSSION TITLE
Question: Can you use saved PaymentMethods for later use in PaymentSheet ?

### DISCUSSION THREAD

**Discussion by sam-parks** [flutter-stripe/flutter_stripe#571] [Q&A]
**Describe the bug**
I am able to initialize the PaymentSheet when I have a PaymentIntent that does not have a customer and payment method associated with it. As soon as I attach a customer and a payment method with the purpose of using saved credit card info, the PaymentSheet fails with a generic error code. 

It states in the documentation that the sheet supports saving and reusing cards. What is the correct way to do this? 

**Expected behavior**
Can use saved credit card info when checking out with the PaymentSheet.

**Smartphone / tablet**
 - Device: Iphone13
 - OS: iOS 15.2
 - Package version: 2.1.1 
 - Flutter version 2.8.0

**Comment by jonasbark** [comment]
Hi Sam!

It works - check out the official documentation for stripe-react-native. Our Flutter plugin is very similar to it:
https://stripe.com/docs/payments/save-and-reuse?platform=react-native&ui=payment-sheet#react-native-collect-payment-details

In essence: use setupIntentClientSecret

**Reply by sam-parks** [reply]
Thank you! Been trying to figure this out for a while now

**Reply by Joel-Borrero** [reply]
Did you get solution for this? I'm trying with paymentIntentClientSecret=response['client_secret'] but there is no loading the payment method, and when creating the PaymentIntent on the server, attaching the saved card_id to the PaymentIntent payment_method parm but on present that sheet, is still asking for card details
